### PR TITLE
Added 3rd Party Software Licenses Page

### LINKED ISSUE #8484: Add third-party Licenses
- [ ] Check for any conflicts in the license for Care and its dependent packages.

Some tools -
- [ ] https://github.com/wrapp/licensecrawler
- [ ] https://github.com/mwittig/npm-license-crawler

### PR DESCRIPTION
**Pull request by tellmeY18**
## Proposed Changes

- Fixes #8484 
- Created SBOM Viewer and License Page
- Added Routes to /licenses and linked it to Login Page 
- setup workflow to generate cyclonedx sbom.json

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

### CODE REVIEW

**Review by rithviknishad — COMMENTED** [review]
Should we keep the generated file git ignored and generate it during the build step instead of using a CI?

cc: @bodhish

File: src/Components/SBOMViewer.tsx [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+  Zlib: "https://opensource.org/licenses/Zlib",
+  "BSL-1.0": "https://opensource.org/licenses/BSL-1.0",
+  "Apache-1.1": "https://opensource.org/licenses/Apache-1.1",
+  "Apache-1.0": "https://opensource.org/licenses/Apache-1.0",
+  "AFL-3.0": "https://opensource.org/licenses/AFL-3.0",
+  "AFL-2.1": "https://opensource.org/licenses/AFL-2.1",
+  "AFL-2.0": "https://opensource.org/licenses/AFL-2.0",
+  "AFL-1.2": "https://opensource.org/licenses/AFL-1.2",
+  "AFL-1.1": "https://opensource.org/licenses/AFL-1.1",
+  "AFL-1.0": "https://opensource.org/licenses/AFL-1.0",
```
Comment (rithviknishad) [inline src/Components/SBOMViewer.tsx]: we could keep this in a JSON file and import that JSON file directly instead.

File: src/Components/SBOMViewer.tsx [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+    axios
+      .get("/bom.json")
+      .then((response) => setBOM(response.data))
+      .catch((error) => console.error("Error fetching BOM:", error));
+
+    axios
+      .get("/be-sbom.json")
+      .then((response) => setBeBOM(response.data))
+      .catch((error) => console.error("Error fetching BE SBOM:", error));
+  }, []);
```
Comment (rithviknishad) [inline src/Components/SBOMViewer.tsx]: use CARE's custom data fetching utility fn. such as `useQuery`/`request` instead.

**Review by rithviknishad — CHANGES_REQUESTED** [review]

File: src/Components/SBOMViewer.tsx (hunk 1 above)
Comment (tellmeY18) [reply src/Components/SBOMViewer.tsx]: done ✅

File: src/Components/SBOMViewer.tsx (hunk 2 above)
Comment (tellmeY18) [reply src/Components/SBOMViewer.tsx]: done ✅

File: src/Components/Licenses/SBOMViewer.tsx [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+    setBeBOM(beBomData as CycloneDXBOM);
+  }, []);
+
+  const handleCopy = () => {
+    setCopyStatus(true);
+    setTimeout(() => setCopyStatus(false), 2000);
+  };
+
+  const renderBOM = (bomData: CycloneDXBOM | null) => {
+    if (!bomData) return <div>Loading BOM...</div>;
```
Comment (rithviknishad) [inline src/Components/Licenses/SBOMViewer.tsx]: can this be made as a separate component instead of inlining the fn.

**Review by rithviknishad — APPROVED** [review]

**Comment by nihal467** [comment]
*Bug*

image

- [ ] the dedicated license page is not mobile-responsive

*UI Enhancement*


image

- [ ] The external reference in the current UI gives the impression that it's empty. I would suggest adding an icon to collapse and expand, so users understand that data exists and they can click on the icon to view it

*UI Enhancement Question*

image

image

@rithviknishad shall we change 3rd to Third, along with adding " | " for separation, would that look better in the UI?


@rithviknishad @tellmeY18

**Comment by nihal467** [comment]
@rithviknishad @tellmeY18 
image

the mobile view is still not fully responsive

**Comment by nihal467** [comment]
LGTM

**Pull request merged**